### PR TITLE
cfg: Update preferred region data on-demand.

### PR DESCRIPTION
**Pull request by Steveice10**
* Remove app loader dependency on the CFG module by storing the preferred region information and querying it from the CFG module on-demand. Should solve issues loading LLE modules before CFG is initialized.
* Simplify areas where the CFG module is used by moving the create-a-new-instance fallback into the `GetModule` helper.

Supercedes https://github.com/citra-emu/citra/pull/6943. Does not have its issues with home menu or using system global instance.

- - -
This change is [Reviewable](https://reviewable.io/reviews/citra-emu/citra/7206)

### CODE REVIEW

File: src/core/hle/service/cfg/cfg.cpp [hunk 1]
```diff
@@ -810,31 +824,30 @@ bool Module::IsSystemSetupNeeded() {
...
+        auto cfg = system.ServiceManager().GetService<Module::Interface>("cfg:u");
+        if (cfg) {
+            return cfg->GetModule();
+        }
+    }
+
+    // If the system is not running or the module is missing,
+    // create an ad-hoc module instance to read data from.
+    return std::make_shared<Module>(system);
 }
```
Comment (GPUCode) [inline src/core/hle/service/cfg/cfg.cpp:837]: Early return condition can be flipped here to reduce nesting

File: src/core/loader/loader.h [hunk 2]
```diff
@@ -91,6 +91,14 @@ class AppLoader : NonCopyable {
      */
     virtual FileType GetFileType() = 0;
 
+    /**
+     * Returns the preferred region codes of this file
+     * @return A vector of the preferred region codes
+     */
+    [[nodiscard]] virtual std::vector<u32> GetPreferredRegions() const {
```
Comment (GPUCode) [inline src/core/loader/loader.h:98]:
```suggestion
    [[nodiscard]] virtual std::span<const u32> GetPreferredRegions() const {
```

**Review by GPUCode — APPROVED** [review]

File: src/core/hle/service/cfg/cfg.cpp (hunk 1 above)
Comment (Steveice10) [reply src/core/hle/service/cfg/cfg.cpp:837]: Can you demonstrate what you have in mind? As the nested condition requires a variable to be initialized. If the conditions were flipped we would need to duplicate this line.

**Review by FearlessTobi — APPROVED** [review]

File: src/core/hle/service/cfg/cfg.cpp (hunk 1 above)
Comment (GPUCode) [reply src/core/hle/service/cfg/cfg.cpp:837]:
```
// If the system is not running or the module is missing,
// create an ad-hoc module instance to read data from.    
if (!system.IsPoweredOn()) {
    return std::make_shared<Module>(system);
}

auto cfg = system.ServiceManager().GetService<Module::Interface>("cfg:u");
if (cfg) {
    return cfg->GetModule();
}
```

Comment (GPUCode) [reply src/core/hle/service/cfg/cfg.cpp:837]: Hm good point, better not duplicate this

**Pull request merged**